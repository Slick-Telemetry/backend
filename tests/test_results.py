# App
from . import client_with_auth


# region good inputs


def test_get_results(snapshot):
    response = client_with_auth.get("/results/2023/5")
    snapshot.assert_match(response.json())


def test_get_results_with_session(snapshot):
    response = client_with_auth.get("/results/2023/5?session=4")
    snapshot.assert_match(response.json())


# endregion good inputs

# region bad inputs


def test_get_results_bad_round_invalid(snapshot):
    response = client_with_auth.get("/results/2023/25?session=2")
    snapshot.assert_match(response.json())


# endregion bad inputs
