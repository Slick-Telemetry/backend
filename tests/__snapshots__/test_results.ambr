# serializer version: 1
# name: test_get_results
  list([
    dict({
      'Abbreviation': 'VER',
      'BroadcastName': 'M VERSTAPPEN',
      'ClassifiedPosition': '1',
      'CountryCode': 'NED',
      'DriverId': 'max_verstappen',
      'DriverNumber': '1',
      'FirstName': 'Max',
      'FullName': 'Max Verstappen',
      'GridPosition': 9.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/M/MAXVER01_Max_Verstappen/maxver01.png.transform/1col/image.png',
      'LastName': 'Verstappen',
      'Points': 26.0,
      'Position': 1.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '3671C6',
      'TeamId': 'red_bull',
      'TeamName': 'Red Bull Racing',
      'Time': 5258241,
    }),
    dict({
      'Abbreviation': 'PER',
      'BroadcastName': 'S PEREZ',
      'ClassifiedPosition': '2',
      'CountryCode': 'MEX',
      'DriverId': 'perez',
      'DriverNumber': '11',
      'FirstName': 'Sergio',
      'FullName': 'Sergio Perez',
      'GridPosition': 1.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/S/SERPER01_Sergio_Perez/serper01.png.transform/1col/image.png',
      'LastName': 'Perez',
      'Points': 18.0,
      'Position': 2.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '3671C6',
      'TeamId': 'red_bull',
      'TeamName': 'Red Bull Racing',
      'Time': 5384,
    }),
    dict({
      'Abbreviation': 'ALO',
      'BroadcastName': 'F ALONSO',
      'ClassifiedPosition': '3',
      'CountryCode': 'ESP',
      'DriverId': 'alonso',
      'DriverNumber': '14',
      'FirstName': 'Fernando',
      'FullName': 'Fernando Alonso',
      'GridPosition': 2.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/F/FERALO01_Fernando_Alonso/feralo01.png.transform/1col/image.png',
      'LastName': 'Alonso',
      'Points': 15.0,
      'Position': 3.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '358C75',
      'TeamId': 'aston_martin',
      'TeamName': 'Aston Martin',
      'Time': 26305,
    }),
    dict({
      'Abbreviation': 'RUS',
      'BroadcastName': 'G RUSSELL',
      'ClassifiedPosition': '4',
      'CountryCode': 'GBR',
      'DriverId': 'russell',
      'DriverNumber': '63',
      'FirstName': 'George',
      'FullName': 'George Russell',
      'GridPosition': 6.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/G/GEORUS01_George_Russell/georus01.png.transform/1col/image.png',
      'LastName': 'Russell',
      'Points': 12.0,
      'Position': 4.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '6CD3BF',
      'TeamId': 'mercedes',
      'TeamName': 'Mercedes',
      'Time': 33229,
    }),
    dict({
      'Abbreviation': 'SAI',
      'BroadcastName': 'C SAINZ',
      'ClassifiedPosition': '5',
      'CountryCode': 'ESP',
      'DriverId': 'sainz',
      'DriverNumber': '55',
      'FirstName': 'Carlos',
      'FullName': 'Carlos Sainz',
      'GridPosition': 3.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/C/CARSAI01_Carlos_Sainz/carsai01.png.transform/1col/image.png',
      'LastName': 'Sainz',
      'Points': 10.0,
      'Position': 5.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'F91536',
      'TeamId': 'ferrari',
      'TeamName': 'Ferrari',
      'Time': 42511,
    }),
    dict({
      'Abbreviation': 'HAM',
      'BroadcastName': 'L HAMILTON',
      'ClassifiedPosition': '6',
      'CountryCode': 'GBR',
      'DriverId': 'hamilton',
      'DriverNumber': '44',
      'FirstName': 'Lewis',
      'FullName': 'Lewis Hamilton',
      'GridPosition': 13.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LEWHAM01_Lewis_Hamilton/lewham01.png.transform/1col/image.png',
      'LastName': 'Hamilton',
      'Points': 8.0,
      'Position': 6.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '6CD3BF',
      'TeamId': 'mercedes',
      'TeamName': 'Mercedes',
      'Time': 51249,
    }),
    dict({
      'Abbreviation': 'LEC',
      'BroadcastName': 'C LECLERC',
      'ClassifiedPosition': '7',
      'CountryCode': 'MON',
      'DriverId': 'leclerc',
      'DriverNumber': '16',
      'FirstName': 'Charles',
      'FullName': 'Charles Leclerc',
      'GridPosition': 7.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/C/CHALEC01_Charles_Leclerc/chalec01.png.transform/1col/image.png',
      'LastName': 'Leclerc',
      'Points': 6.0,
      'Position': 7.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'F91536',
      'TeamId': 'ferrari',
      'TeamName': 'Ferrari',
      'Time': 52988,
    }),
    dict({
      'Abbreviation': 'GAS',
      'BroadcastName': 'P GASLY',
      'ClassifiedPosition': '8',
      'CountryCode': 'FRA',
      'DriverId': 'gasly',
      'DriverNumber': '10',
      'FirstName': 'Pierre',
      'FullName': 'Pierre Gasly',
      'GridPosition': 5.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/P/PIEGAS01_Pierre_Gasly/piegas01.png.transform/1col/image.png',
      'LastName': 'Gasly',
      'Points': 4.0,
      'Position': 8.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '2293D1',
      'TeamId': 'alpine',
      'TeamName': 'Alpine',
      'Time': 55670,
    }),
    dict({
      'Abbreviation': 'OCO',
      'BroadcastName': 'E OCON',
      'ClassifiedPosition': '9',
      'CountryCode': 'FRA',
      'DriverId': 'ocon',
      'DriverNumber': '31',
      'FirstName': 'Esteban',
      'FullName': 'Esteban Ocon',
      'GridPosition': 8.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/E/ESTOCO01_Esteban_Ocon/estoco01.png.transform/1col/image.png',
      'LastName': 'Ocon',
      'Points': 2.0,
      'Position': 9.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '2293D1',
      'TeamId': 'alpine',
      'TeamName': 'Alpine',
      'Time': 58123,
    }),
    dict({
      'Abbreviation': 'MAG',
      'BroadcastName': 'K MAGNUSSEN',
      'ClassifiedPosition': '10',
      'CountryCode': 'DEN',
      'DriverId': 'kevin_magnussen',
      'DriverNumber': '20',
      'FirstName': 'Kevin',
      'FullName': 'Kevin Magnussen',
      'GridPosition': 4.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/K/KEVMAG01_Kevin_Magnussen/kevmag01.png.transform/1col/image.png',
      'LastName': 'Magnussen',
      'Points': 1.0,
      'Position': 10.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'B6BABD',
      'TeamId': 'haas',
      'TeamName': 'Haas F1 Team',
      'Time': 62945,
    }),
    dict({
      'Abbreviation': 'TSU',
      'BroadcastName': 'Y TSUNODA',
      'ClassifiedPosition': '11',
      'CountryCode': 'JPN',
      'DriverId': 'tsunoda',
      'DriverNumber': '22',
      'FirstName': 'Yuki',
      'FullName': 'Yuki Tsunoda',
      'GridPosition': 17.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/Y/YUKTSU01_Yuki_Tsunoda/yuktsu01.png.transform/1col/image.png',
      'LastName': 'Tsunoda',
      'Points': 0.0,
      'Position': 11.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '5E8FAA',
      'TeamId': 'alphatauri',
      'TeamName': 'AlphaTauri',
      'Time': 64309,
    }),
    dict({
      'Abbreviation': 'STR',
      'BroadcastName': 'L STROLL',
      'ClassifiedPosition': '12',
      'CountryCode': 'CAN',
      'DriverId': 'stroll',
      'DriverNumber': '18',
      'FirstName': 'Lance',
      'FullName': 'Lance Stroll',
      'GridPosition': 18.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LANSTR01_Lance_Stroll/lanstr01.png.transform/1col/image.png',
      'LastName': 'Stroll',
      'Points': 0.0,
      'Position': 12.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '358C75',
      'TeamId': 'aston_martin',
      'TeamName': 'Aston Martin',
      'Time': 64754,
    }),
    dict({
      'Abbreviation': 'BOT',
      'BroadcastName': 'V BOTTAS',
      'ClassifiedPosition': '13',
      'CountryCode': 'FIN',
      'DriverId': 'bottas',
      'DriverNumber': '77',
      'FirstName': 'Valtteri',
      'FullName': 'Valtteri Bottas',
      'GridPosition': 10.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/V/VALBOT01_Valtteri_Bottas/valbot01.png.transform/1col/image.png',
      'LastName': 'Bottas',
      'Points': 0.0,
      'Position': 13.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'C92D4B',
      'TeamId': 'alfa',
      'TeamName': 'Alfa Romeo',
      'Time': 71637,
    }),
    dict({
      'Abbreviation': 'ALB',
      'BroadcastName': 'A ALBON',
      'ClassifiedPosition': '14',
      'CountryCode': 'THA',
      'DriverId': 'albon',
      'DriverNumber': '23',
      'FirstName': 'Alexander',
      'FullName': 'Alexander Albon',
      'GridPosition': 11.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/A/ALEALB01_Alexander_Albon/alealb01.png.transform/1col/image.png',
      'LastName': 'Albon',
      'Points': 0.0,
      'Position': 14.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '37BEDD',
      'TeamId': 'williams',
      'TeamName': 'Williams',
      'Time': 72861,
    }),
    dict({
      'Abbreviation': 'HUL',
      'BroadcastName': 'N HULKENBERG',
      'ClassifiedPosition': '15',
      'CountryCode': 'GER',
      'DriverId': 'hulkenberg',
      'DriverNumber': '27',
      'FirstName': 'Nico',
      'FullName': 'Nico Hulkenberg',
      'GridPosition': 12.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/N/NICHUL01_Nico_Hulkenberg/nichul01.png.transform/1col/image.png',
      'LastName': 'Hulkenberg',
      'Points': 0.0,
      'Position': 15.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'B6BABD',
      'TeamId': 'haas',
      'TeamName': 'Haas F1 Team',
      'Time': 74950,
    }),
    dict({
      'Abbreviation': 'ZHO',
      'BroadcastName': 'G ZHOU',
      'ClassifiedPosition': '16',
      'CountryCode': 'CHN',
      'DriverId': 'zhou',
      'DriverNumber': '24',
      'FirstName': 'Guanyu',
      'FullName': 'Guanyu Zhou',
      'GridPosition': 14.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/G/GUAZHO01_Guanyu_Zhou/guazho01.png.transform/1col/image.png',
      'LastName': 'Zhou',
      'Points': 0.0,
      'Position': 16.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'C92D4B',
      'TeamId': 'alfa',
      'TeamName': 'Alfa Romeo',
      'Time': 78440,
    }),
    dict({
      'Abbreviation': 'NOR',
      'BroadcastName': 'L NORRIS',
      'ClassifiedPosition': '17',
      'CountryCode': 'GBR',
      'DriverId': 'norris',
      'DriverNumber': '4',
      'FirstName': 'Lando',
      'FullName': 'Lando Norris',
      'GridPosition': 16.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LANNOR01_Lando_Norris/lannor01.png.transform/1col/image.png',
      'LastName': 'Norris',
      'Points': 0.0,
      'Position': 17.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': 'F58020',
      'TeamId': 'mclaren',
      'TeamName': 'McLaren',
      'Time': 87717,
    }),
    dict({
      'Abbreviation': 'DEV',
      'BroadcastName': 'N DE VRIES',
      'ClassifiedPosition': '18',
      'CountryCode': 'NED',
      'DriverId': 'de_vries',
      'DriverNumber': '21',
      'FirstName': 'Nyck',
      'FullName': 'Nyck De Vries',
      'GridPosition': 15.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/N/NYCDEV01_Nyck_De%20Vries/nycdev01.png.transform/1col/image.png',
      'LastName': 'De Vries',
      'Points': 0.0,
      'Position': 18.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': 'Finished',
      'TeamColor': '5E8FAA',
      'TeamId': 'alphatauri',
      'TeamName': 'AlphaTauri',
      'Time': 88949,
    }),
    dict({
      'Abbreviation': 'PIA',
      'BroadcastName': 'O PIASTRI',
      'ClassifiedPosition': '19',
      'CountryCode': 'AUS',
      'DriverId': 'piastri',
      'DriverNumber': '81',
      'FirstName': 'Oscar',
      'FullName': 'Oscar Piastri',
      'GridPosition': 19.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/O/OSCPIA01_Oscar_Piastri/oscpia01.png.transform/1col/image.png',
      'LastName': 'Piastri',
      'Points': 0.0,
      'Position': 19.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': '+1 Lap',
      'TeamColor': 'F58020',
      'TeamId': 'mclaren',
      'TeamName': 'McLaren',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'SAR',
      'BroadcastName': 'L SARGEANT',
      'ClassifiedPosition': '20',
      'CountryCode': 'USA',
      'DriverId': 'sargeant',
      'DriverNumber': '2',
      'FirstName': 'Logan',
      'FullName': 'Logan Sargeant',
      'GridPosition': 20.0,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LOGSAR01_Logan_Sargeant/logsar01.png.transform/1col/image.png',
      'LastName': 'Sargeant',
      'Points': 0.0,
      'Position': 20.0,
      'Q1': None,
      'Q2': None,
      'Q3': None,
      'Status': '+1 Lap',
      'TeamColor': '37BEDD',
      'TeamId': 'williams',
      'TeamName': 'Williams',
      'Time': None,
    }),
  ])
# ---
# name: test_get_results_bad_round_invalid
  dict({
    'detail': 'Bad Request. Invalid round: 25',
  })
# ---
# name: test_get_results_with_session
  list([
    dict({
      'Abbreviation': 'PER',
      'BroadcastName': 'S PEREZ',
      'ClassifiedPosition': '',
      'CountryCode': 'MEX',
      'DriverId': 'perez',
      'DriverNumber': '11',
      'FirstName': 'Sergio',
      'FullName': 'Sergio Perez',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/S/SERPER01_Sergio_Perez/serper01.png.transform/1col/image.png',
      'LastName': 'Perez',
      'Points': None,
      'Position': 1.0,
      'Q1': 87713,
      'Q2': 87328,
      'Q3': 86841,
      'Status': '',
      'TeamColor': '3671C6',
      'TeamId': 'red_bull',
      'TeamName': 'Red Bull Racing',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'ALO',
      'BroadcastName': 'F ALONSO',
      'ClassifiedPosition': '',
      'CountryCode': 'ESP',
      'DriverId': 'alonso',
      'DriverNumber': '14',
      'FirstName': 'Fernando',
      'FullName': 'Fernando Alonso',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/F/FERALO01_Fernando_Alonso/feralo01.png.transform/1col/image.png',
      'LastName': 'Alonso',
      'Points': None,
      'Position': 2.0,
      'Q1': 88179,
      'Q2': 87097,
      'Q3': 87202,
      'Status': '',
      'TeamColor': '358C75',
      'TeamId': 'aston_martin',
      'TeamName': 'Aston Martin',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'SAI',
      'BroadcastName': 'C SAINZ',
      'ClassifiedPosition': '',
      'CountryCode': 'ESP',
      'DriverId': 'sainz',
      'DriverNumber': '55',
      'FirstName': 'Carlos',
      'FullName': 'Carlos Sainz',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/C/CARSAI01_Carlos_Sainz/carsai01.png.transform/1col/image.png',
      'LastName': 'Sainz',
      'Points': None,
      'Position': 3.0,
      'Q1': 87686,
      'Q2': 87148,
      'Q3': 87349,
      'Status': '',
      'TeamColor': 'F91536',
      'TeamId': 'ferrari',
      'TeamName': 'Ferrari',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'MAG',
      'BroadcastName': 'K MAGNUSSEN',
      'ClassifiedPosition': '',
      'CountryCode': 'DEN',
      'DriverId': 'kevin_magnussen',
      'DriverNumber': '20',
      'FirstName': 'Kevin',
      'FullName': 'Kevin Magnussen',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/K/KEVMAG01_Kevin_Magnussen/kevmag01.png.transform/1col/image.png',
      'LastName': 'Magnussen',
      'Points': None,
      'Position': 4.0,
      'Q1': 87809,
      'Q2': 87673,
      'Q3': 87767,
      'Status': '',
      'TeamColor': 'B6BABD',
      'TeamId': 'haas',
      'TeamName': 'Haas F1 Team',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'GAS',
      'BroadcastName': 'P GASLY',
      'ClassifiedPosition': '',
      'CountryCode': 'FRA',
      'DriverId': 'gasly',
      'DriverNumber': '10',
      'FirstName': 'Pierre',
      'FullName': 'Pierre Gasly',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/P/PIEGAS01_Pierre_Gasly/piegas01.png.transform/1col/image.png',
      'LastName': 'Gasly',
      'Points': None,
      'Position': 5.0,
      'Q1': 88061,
      'Q2': 87612,
      'Q3': 87786,
      'Status': '',
      'TeamColor': '2293D1',
      'TeamId': 'alpine',
      'TeamName': 'Alpine',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'RUS',
      'BroadcastName': 'G RUSSELL',
      'ClassifiedPosition': '',
      'CountryCode': 'GBR',
      'DriverId': 'russell',
      'DriverNumber': '63',
      'FirstName': 'George',
      'FullName': 'George Russell',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/G/GEORUS01_George_Russell/georus01.png.transform/1col/image.png',
      'LastName': 'Russell',
      'Points': None,
      'Position': 6.0,
      'Q1': 88086,
      'Q2': 87743,
      'Q3': 87804,
      'Status': '',
      'TeamColor': '6CD3BF',
      'TeamId': 'mercedes',
      'TeamName': 'Mercedes',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'LEC',
      'BroadcastName': 'C LECLERC',
      'ClassifiedPosition': '',
      'CountryCode': 'MON',
      'DriverId': 'leclerc',
      'DriverNumber': '16',
      'FirstName': 'Charles',
      'FullName': 'Charles Leclerc',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/C/CHALEC01_Charles_Leclerc/chalec01.png.transform/1col/image.png',
      'LastName': 'Leclerc',
      'Points': None,
      'Position': 7.0,
      'Q1': 87713,
      'Q2': 86964,
      'Q3': 87861,
      'Status': '',
      'TeamColor': 'F91536',
      'TeamId': 'ferrari',
      'TeamName': 'Ferrari',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'OCO',
      'BroadcastName': 'E OCON',
      'ClassifiedPosition': '',
      'CountryCode': 'FRA',
      'DriverId': 'ocon',
      'DriverNumber': '31',
      'FirstName': 'Esteban',
      'FullName': 'Esteban Ocon',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/E/ESTOCO01_Esteban_Ocon/estoco01.png.transform/1col/image.png',
      'LastName': 'Ocon',
      'Points': None,
      'Position': 8.0,
      'Q1': 87872,
      'Q2': 87444,
      'Q3': 87935,
      'Status': '',
      'TeamColor': '2293D1',
      'TeamId': 'alpine',
      'TeamName': 'Alpine',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'VER',
      'BroadcastName': 'M VERSTAPPEN',
      'ClassifiedPosition': '',
      'CountryCode': 'NED',
      'DriverId': 'max_verstappen',
      'DriverNumber': '1',
      'FirstName': 'Max',
      'FullName': 'Max Verstappen',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/M/MAXVER01_Max_Verstappen/maxver01.png.transform/1col/image.png',
      'LastName': 'Verstappen',
      'Points': None,
      'Position': 9.0,
      'Q1': 87363,
      'Q2': 86814,
      'Q3': None,
      'Status': '',
      'TeamColor': '3671C6',
      'TeamId': 'red_bull',
      'TeamName': 'Red Bull Racing',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'BOT',
      'BroadcastName': 'V BOTTAS',
      'ClassifiedPosition': '',
      'CountryCode': 'FIN',
      'DriverId': 'bottas',
      'DriverNumber': '77',
      'FirstName': 'Valtteri',
      'FullName': 'Valtteri Bottas',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/V/VALBOT01_Valtteri_Bottas/valbot01.png.transform/1col/image.png',
      'LastName': 'Bottas',
      'Points': None,
      'Position': 10.0,
      'Q1': 87864,
      'Q2': 87564,
      'Q3': None,
      'Status': '',
      'TeamColor': 'C92D4B',
      'TeamId': 'alfa',
      'TeamName': 'Alfa Romeo',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'ALB',
      'BroadcastName': 'A ALBON',
      'ClassifiedPosition': '',
      'CountryCode': 'THA',
      'DriverId': 'albon',
      'DriverNumber': '23',
      'FirstName': 'Alexander',
      'FullName': 'Alexander Albon',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/A/ALEALB01_Alexander_Albon/alealb01.png.transform/1col/image.png',
      'LastName': 'Albon',
      'Points': None,
      'Position': 11.0,
      'Q1': 88234,
      'Q2': 87795,
      'Q3': None,
      'Status': '',
      'TeamColor': '37BEDD',
      'TeamId': 'williams',
      'TeamName': 'Williams',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'HUL',
      'BroadcastName': 'N HULKENBERG',
      'ClassifiedPosition': '',
      'CountryCode': 'GER',
      'DriverId': 'hulkenberg',
      'DriverNumber': '27',
      'FirstName': 'Nico',
      'FullName': 'Nico Hulkenberg',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/N/NICHUL01_Nico_Hulkenberg/nichul01.png.transform/1col/image.png',
      'LastName': 'Hulkenberg',
      'Points': None,
      'Position': 12.0,
      'Q1': 87945,
      'Q2': 87903,
      'Q3': None,
      'Status': '',
      'TeamColor': 'B6BABD',
      'TeamId': 'haas',
      'TeamName': 'Haas F1 Team',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'HAM',
      'BroadcastName': 'L HAMILTON',
      'ClassifiedPosition': '',
      'CountryCode': 'GBR',
      'DriverId': 'hamilton',
      'DriverNumber': '44',
      'FirstName': 'Lewis',
      'FullName': 'Lewis Hamilton',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LEWHAM01_Lewis_Hamilton/lewham01.png.transform/1col/image.png',
      'LastName': 'Hamilton',
      'Points': None,
      'Position': 13.0,
      'Q1': 87846,
      'Q2': 87975,
      'Q3': None,
      'Status': '',
      'TeamColor': '6CD3BF',
      'TeamId': 'mercedes',
      'TeamName': 'Mercedes',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'ZHO',
      'BroadcastName': 'G ZHOU',
      'ClassifiedPosition': '',
      'CountryCode': 'CHN',
      'DriverId': 'zhou',
      'DriverNumber': '24',
      'FirstName': 'Guanyu',
      'FullName': 'Guanyu Zhou',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/G/GUAZHO01_Guanyu_Zhou/guazho01.png.transform/1col/image.png',
      'LastName': 'Zhou',
      'Points': None,
      'Position': 14.0,
      'Q1': 88180,
      'Q2': 88091,
      'Q3': None,
      'Status': '',
      'TeamColor': 'C92D4B',
      'TeamId': 'alfa',
      'TeamName': 'Alfa Romeo',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'DEV',
      'BroadcastName': 'N DE VRIES',
      'ClassifiedPosition': '',
      'CountryCode': 'NED',
      'DriverId': 'de_vries',
      'DriverNumber': '21',
      'FirstName': 'Nyck',
      'FullName': 'Nyck De Vries',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/N/NYCDEV01_Nyck_De%20Vries/nycdev01.png.transform/1col/image.png',
      'LastName': 'De Vries',
      'Points': None,
      'Position': 15.0,
      'Q1': 88325,
      'Q2': 88395,
      'Q3': None,
      'Status': '',
      'TeamColor': '5E8FAA',
      'TeamId': 'alphatauri',
      'TeamName': 'AlphaTauri',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'NOR',
      'BroadcastName': 'L NORRIS',
      'ClassifiedPosition': '',
      'CountryCode': 'GBR',
      'DriverId': 'norris',
      'DriverNumber': '4',
      'FirstName': 'Lando',
      'FullName': 'Lando Norris',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LANNOR01_Lando_Norris/lannor01.png.transform/1col/image.png',
      'LastName': 'Norris',
      'Points': None,
      'Position': 16.0,
      'Q1': 88394,
      'Q2': None,
      'Q3': None,
      'Status': '',
      'TeamColor': 'F58020',
      'TeamId': 'mclaren',
      'TeamName': 'McLaren',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'TSU',
      'BroadcastName': 'Y TSUNODA',
      'ClassifiedPosition': '',
      'CountryCode': 'JPN',
      'DriverId': 'tsunoda',
      'DriverNumber': '22',
      'FirstName': 'Yuki',
      'FullName': 'Yuki Tsunoda',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/Y/YUKTSU01_Yuki_Tsunoda/yuktsu01.png.transform/1col/image.png',
      'LastName': 'Tsunoda',
      'Points': None,
      'Position': 17.0,
      'Q1': 88429,
      'Q2': None,
      'Q3': None,
      'Status': '',
      'TeamColor': '5E8FAA',
      'TeamId': 'alphatauri',
      'TeamName': 'AlphaTauri',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'STR',
      'BroadcastName': 'L STROLL',
      'ClassifiedPosition': '',
      'CountryCode': 'CAN',
      'DriverId': 'stroll',
      'DriverNumber': '18',
      'FirstName': 'Lance',
      'FullName': 'Lance Stroll',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LANSTR01_Lance_Stroll/lanstr01.png.transform/1col/image.png',
      'LastName': 'Stroll',
      'Points': None,
      'Position': 18.0,
      'Q1': 88476,
      'Q2': None,
      'Q3': None,
      'Status': '',
      'TeamColor': '358C75',
      'TeamId': 'aston_martin',
      'TeamName': 'Aston Martin',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'PIA',
      'BroadcastName': 'O PIASTRI',
      'ClassifiedPosition': '',
      'CountryCode': 'AUS',
      'DriverId': 'piastri',
      'DriverNumber': '81',
      'FirstName': 'Oscar',
      'FullName': 'Oscar Piastri',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/O/OSCPIA01_Oscar_Piastri/oscpia01.png.transform/1col/image.png',
      'LastName': 'Piastri',
      'Points': None,
      'Position': 19.0,
      'Q1': 88484,
      'Q2': None,
      'Q3': None,
      'Status': '',
      'TeamColor': 'F58020',
      'TeamId': 'mclaren',
      'TeamName': 'McLaren',
      'Time': None,
    }),
    dict({
      'Abbreviation': 'SAR',
      'BroadcastName': 'L SARGEANT',
      'ClassifiedPosition': '',
      'CountryCode': 'USA',
      'DriverId': 'sargeant',
      'DriverNumber': '2',
      'FirstName': 'Logan',
      'FullName': 'Logan Sargeant',
      'GridPosition': None,
      'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/L/LOGSAR01_Logan_Sargeant/logsar01.png.transform/1col/image.png',
      'LastName': 'Sargeant',
      'Points': None,
      'Position': 20.0,
      'Q1': 88577,
      'Q2': None,
      'Q3': None,
      'Status': '',
      'TeamColor': '37BEDD',
      'TeamId': 'williams',
      'TeamName': 'Williams',
      'Time': None,
    }),
  ])
# ---
